app/index.ts

```diff
@@ -27,9 +27,12 @@ Encoder.BUFFER_SIZE = 512 * 1024
 
 async function main() {
   if (process.env.NODE_APP_INSTANCE) {
+    const processNumber = Number(process.env.NODE_APP_INSTANCE ?? "0")
+    const port = (Number(process.env.PORT) ?? 2567) + processNumber
```

Comment:
Should we guard against PORT being set to a non-numeric value? Number("abc") would yield NaN, and NaN ?? 2567 still results in NaN, so the fallback wouldn’t kick in.
`const basePort = Number(process.env.PORT)`
`const port = (!isNaN(basePort) ? basePort : 2567) + processNumber` 
could work better

Reply:
lets assume we configure environment variables correctly